packages/cli/src/modules/mcp/tools/utils.ts

```diff
@@ -0,0 +1,145 @@
+import type { User } from '@n8n/db';
+import type { INode } from 'n8n-workflow';
+
+import {
+	hasHttpHeaderAuthDecryptedData,
+	hasJwtPemKeyDecryptedData,
+	hasJwtSecretDecryptedData,
+} from '../mcp.typeguards';
+
+import type { CredentialsService } from '@/credentials/credentials.service';
+
+export const getWebhookDetails = async (
+	user: User,
+	webhookNodes: INode[],
+	baseUrl: string,
+	isWorkflowActive: boolean,
+	crednetialsService: CredentialsService,
+): Promise<string> => {
+	if (webhookNodes.length === 0) {
+		return 'This workflow does not have a trigger node that can be executed via MCP.';
+	}
+
+	let triggerNotice = 'This workflow is triggered by the following webhook(s):\n\n';
+
+	const webhookPromises = webhookNodes.map(async (node, index) => {
+		let credentialsInfo: string | null = null;
+		if (node.parameters.authentication) {
+			const authType =
+				typeof node.parameters.authentication === 'string'
+					? node.parameters.authentication
+					: undefined;
+			switch (authType) {
+				case 'basicAuth':
+					credentialsInfo =
+						'\n\t - This webhook requires basic authentication with a username and password that should be provided by the user.';
+					break;
+				case 'headerAuth': {
+					const headerAuthDetails = await getHeaderAuthDetails(user, node, crednetialsService);
+					if (headerAuthDetails) {
+						credentialsInfo = headerAuthDetails;
+					}
+					break;
+				}
+				case 'jwtAuth': {
+					const jwtDetails = await getJWTAuthDetails(user, node, crednetialsService);
+					if (jwtDetails) {
+						credentialsInfo = jwtDetails;
+					}
+					break;
+				}
+			}
+		}
+
+		const responseModeDetails = getResponseModeDescription(node);
+		const pathParam = typeof node.parameters.path === 'string' ? node.parameters.path : '';
+		const httpMethod =
+			typeof node.parameters.httpMethod === 'string' ? node.parameters.httpMethod : 'GET';
+
+		return `
+				<trigger ${index + 1}>
+				\t - Node name: ${node.name}
+				\t - Base URL: ${baseUrl}
+				\t - PATH: ${isWorkflowActive ? '/webhook/' : '/webhook-test/'}${pathParam}
+				\t - HTTP Method: ${httpMethod}
+				\t - Response Mode: ${responseModeDetails}
+				${
+					credentialsInfo
+						? `\t - Credentials: ${credentialsInfo}`
+						: '\t - No credentials required for this webhook.'
+				}
+				</trigger ${index + 1}>`;
+	});
+
+	const webhookResults = await Promise.all(webhookPromises);
+	triggerNotice += webhookResults.join('\n\n');
+
+	return triggerNotice;
+};
+
+const getHeaderAuthDetails = async (
+	user: User,
+	node: INode,
+	credentialsService: CredentialsService,
+): Promise<string | null> => {
+	const id = node.credentials?.httpHeaderAuth?.id;
+	if (!id) return null;
+
+	const creds = await credentialsService.getOne(user, id, true);
```

Comment:
Can we avoid decrypting the actual credential here? I feel very unease about accessing the creds just to check what type it is